pkg/sql/conn_executor_exec.go

```diff
@@ -2749,6 +2769,15 @@ func (ex *connExecutor) dispatchReadCommittedStmtToExecutionEngine(
 			ex.sessionTracing.TraceRetryInformation(
 				ctx, "statement", p.autoRetryStmtCounter, p.autoRetryStmtReason,
 			)
+			// Step both the sequence number and the external read timestamp so that
+			// we can see the results of the conflicting transactions that caused us
+			// to fail.
```

Comment:
What do we mean by the word `external` here?

Have you definitely confirmed that without the Step the next statement happens at the same read timestamp? I ask because over in the txn coord sender, when we hit a retriable error we bump the txn coord senders read timestamp to the earliest write timestamp that the write should have succeed at.  So I'm wondering if (1) we have a bug or I haven't follow that code correctly or (2) if the main impact of this is pushing our timestamp a bit further into the future making it more likely we succeed.

Reply:
There are a few mentions of "external" in #108735.

Reply:
Ah, I think I got "external" from this comment, meaning "in relationship to other transactions": https://github.com/cockroachdb/cockroach/blame/680f1757c59f11f69fb04e5fa004cd819800bcf3/pkg/sql/conn_executor_exec.go#L2563-L2574

But confusingly, in the PR Yahor linked, and in other comments in the same file (such as https://github.com/cockroachdb/cockroach/blame/680f1757c59f11f69fb04e5fa004cd819800bcf3/pkg/sql/conn_executor_exec.go#L2046-L2052) "external" actually means "not internal executor"... so there are two different meanings...

I'm going to remove "external" from my comment and from Nathan's comment, since it's not clear this first meaning is used anywhere else.